app/routes.py

```diff
@@ -1,2 +1,245 @@
+        "description" : task.description,
+        "is_complete" : task.is_complete
+    }}), 200
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_tasks_incomplete(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return jsonify(None), 404
+
+    # request_body = request.get_json()
+
+    task.is_complete = False
+    task.completed_at = None
+
+    return jsonify({ "task" : {
+        "id" : task.task_id,
+        "title" : task.title,
+        "description" : task.description,
+        "is_complete" : task.is_complete
+    }}), 200
+
+@goals_bp.route("", methods=["GET","POST"])
+def handle_goals():
+    if request.method == "GET":
+        goals = Goal.query.all()
+
+        goals_response = []
+        for goal in goals:
+            goals_response.append({
+                "id" : goal.goal_id,
+                "title" : goal.title
+            })
+        return jsonify(goals_response)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        title = request_body.get("title")
+
+        if not title:
+            return jsonify({
+                "details" : "Invalid data"
+            }), 400
+
+        new_goal = Goal(title=title)
+
+        db.session.add(new_goal)
+        db.session.commit()
+
+        return jsonify({"goal" : {
+            "id" : new_goal.goal_id,
+            "title" : new_goal.title
+        }}), 201
+
+@goals_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"])
+def handle_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+
+    if not goal:
+        return jsonify(None), 404
+    
+    if request.method == "GET":
+        return jsonify({"goal" : {
+            "id" : goal.goal_id,
+            "title" : goal.title
+        }})
+    
+    elif request.method == "PUT":
+        request_body = request.get_json()
+        title = request_body.get("title")
+
+        if title:
+            goal.title = title
+
+            db.session.commit()
+
+            return jsonify({"goal" : {
+                "id" : goal.goal_id,
+                "title" : goal.title
+            }})
+        else:
+            return jsonify(None), 404
+    
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+
+        return jsonify({
+            "details" : f'Goal {goal.goal_id} "{goal.title}" successfully deleted'
+        }) 
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST", "GET"])
+def handle_goals_tasks(goal_id):
+    goal = Goal.query.get(goal_id)
+    if goal is None:
+        return jsonify(None), 404
+
+    if request.method == "POST":
+        request_body = request.get_json()
+
+        task_ids = request_body.get("task_ids")
+        for task_id in task_ids:
+            task = Task.query.get(task_id)
+            task.goal = goal
+
+        db.session.commit()
+
+        return jsonify({
+            "id" : goal.goal_id,
+            "task_ids" : task_ids
```

Comment:
If our desired behavior for this POST is to add the posted tasks to the existing list of tasks, then we should read the task ids from the updated goal itself rather than from the request.